project_task_category/models/project_task_category.py

```diff
@@ -11,4 +11,5 @@ class ProjectTask(models.Model):
     categ_id = fields.Many2one(
         'project.category',
         'Category',
+        domain=[('code', '=', 'categ_id')]
```

Comment:
Can we have domain listing the child Categories form a predefined root?
You can copy the implementation done for the `categ_ids` Tags, here: 
https://github.com/OCA/project/blob/8.0/project_categ/project_categ_model.py#L83